pkg/kubectl/cluster.go

```diff
@@ -30,6 +31,8 @@ import (
 type ClusterGeneratorV1Beta1 struct {
 	// Name of the cluster context (required)
 	Name string
+	// Name of the federation this cluster is joining (required)
```

Comment:
Adding a new required param breaks existing scripts. Is this really required?

I dont see any value of adding this annotation on cluster objects. Running `kubectl get clusters <cluster-name> -o yaml` will not give me a reusable yaml file that I can use with another federation.

IIUC the original issue was because we store some federation objects in host cluster and since the host cluster can be hosting multiple federation objects it makes sense to add these annotation there.
Cluster objects are already stored in federation control plane, so skip the annotation for clusters?